spinalcordtoolbox/scripts/sct_propseg.py

```diff
@@ -628,14 +631,18 @@ def propseg(img_input, options_dict):
                1, 'error')
         sys.exit(1)
 
-    # rename output files
-    fname_seg_old = os.path.join(folder_output, add_suffix(os.path.basename(fname_data_propseg), "_seg"))
+    # rename output files and move them from the tmp directory to the output directory
+    fname_seg_old = os.path.join(folder_output_tmp, add_suffix(os.path.basename(fname_data_propseg), "_seg"))
     fname_seg = os.path.join(folder_output, fname_out)
     mv(fname_seg_old, fname_seg)
```

Comment:
Well what do you know:

```
_________ test_sct_propseg_check_dice_coefficient_against_groundtruth __________
Traceback (most recent call last):
  File "/__w/spinalcordtoolbox/spinalcordtoolbox/testing/cli/test_cli_sct_propseg.py", line 18, in test_sct_propseg_check_dice_coefficient_against_groundtruth
    sct_propseg.main(argv=['-i', 't2/t2.nii.gz', '-c', 't2', '-qc', 'testing-qc'])
  File "/__w/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_propseg.py", line 686, in main
    img_seg = propseg(img_input, arguments)
  File "/__w/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_propseg.py", line 637, in propseg
    mv(fname_seg_old, fname_seg)
  File "/__w/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/utils/fs.py", line 226, in mv
    os.rename(src, dst)
OSError: [Errno 18] Invalid cross-device link: '/tmp/sct-20220720161823.048907-propseg-3aqvvl9q/t2_seg.nii.gz' -> './t2_seg.nii.gz'
=========================== short test summary info ============================
FAILED testing/cli/test_cli_sct_propseg.py::test_sct_propseg_check_dice_coefficient_against_groundtruth
==== 1 failed, 402 passed, 11 skipped, 89126 warnings in 358.67s (0:05:58) =====
```

Looks like a repeat of https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3832, just in a different context.

Our use of `mv` functions isn't quite consistent; in some places, we use `spinalcordtoolbox.utils.mv`, and in others, we use `shutil.move`. Maybe we should consider trying to unify these calls so that fixes like https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3833 are applied across the board?

Reply:
I've opened https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3848 for the `mv` related changes. I'll update this PR accordingly so that it's propseg-only. :)